lib/api/route/messages.dart

```diff
@@ -1,11 +1,89 @@
 import 'package:json_annotation/json_annotation.dart';
 
 import '../core.dart';
+import '../exception.dart';
 import '../model/model.dart';
 import '../model/narrow.dart';
 
 part 'messages.g.dart';
 
+/// Convenience function to get a single message from any server.
+///
+/// This encapsulates a server-feature check.
+///
+/// Gives null if the server reports that the message doesn't exist.
+// TODO(server-5) Simplify this away; just use getMessage.
+Future<Message?> getMessageCompat(ApiConnection connection, {
+  required int messageId,
+  bool? applyMarkdown,
+}) async {
+  final useLegacyApi = connection.zulipFeatureLevel! < 120;
+  if (useLegacyApi) {
+    final response = await getMessages(connection,
+      narrow: [ApiNarrowMessageId(messageId)],
+      anchor: NumericAnchor(messageId),
+      numBefore: 0,
+      numAfter: 0,
+      applyMarkdown: applyMarkdown,
+
+      // Hard-code this param to `true`, as the new single-message API
+      // effectively does:
+      //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60client_gravatar.60.20in.20.60messages.2F.7Bmessage_id.7D.60/near/1418337
+      clientGravatar: true,
+    );
+    final messages = response.messages;
+    if (messages.isEmpty) {
+      return null;
+    }
+    return messages.first;
+  } else {
+    try {
+      final response = await getMessage(connection,
+        messageId: messageId,
+        applyMarkdown: applyMarkdown,
+      );
+      // message property present at this feature level
+      return response.message!;
+    } on ZulipApiException catch (e) {
+      if (e.code == 'BAD_REQUEST') {
+        // Servers use this code when the message doesn't exist, according to
+        // the example in the doc.
+        return null;
+      }
+      rethrow;
+    }
+  }
+}
+
+/// https://zulip.com/api/get-message
+Future<GetMessageResult> getMessage(ApiConnection connection, {
+  required int messageId,
+  bool? applyMarkdown,
+}) {
+  final supportsApplyMarkdown = connection.zulipFeatureLevel! >= 120; // TODO(server-5)
+  assert(supportsApplyMarkdown || applyMarkdown == null);
+  return connection.get('getMessage', GetMessageResult.fromJson, 'messages/$messageId', {
+    'message_id': messageId,
+    if (applyMarkdown != null) 'apply_markdown': applyMarkdown,
+  });
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class GetMessageResult {
+  final String? rawContent; // (Marked optional for a future where this is removed)
+  final Message? message; // TODO(server-5)
```

Comment:
Since `rawContent` is deprecated and we don't intend to use it, this endpoint doesn't have a remaining use case for older servers where this `message` property is absent.

Given that, I think we can simplify things a bit by effectively pretending that the pre-120 version of this endpoint doesn't exist. That means:
* Have `Message message`, unconditionally.
* Have `getMessage` assert that the feature level is at least 120.

(The latter step isn't strictly necessary — without it we'd get an error anyway — but could help clarify things a bit.)

Then that also means `getMessageCompat` could just `return result.message` without `!`.